skunk/rust/skunk/src/main.rs

```diff
@@ -120,15 +132,31 @@ impl FileData {
   
     let dependencies = ast::uses(&ast);
     let mut processed_modules = Vec::new();
+    let mut newtypes: Vec<ast::NewType> = ast::newtypes(&ast).iter().map(|a| (*a).clone()).collect();
```

Comment:
Nit: I think

.iter().cloned().collect()

Will do the same thing here but be a bit cleaner.

Reply:
almost - I had to switch drain(..) instead because newtypes returns Vec<&NewType>, so newtypes(&ast).iter() was Iter<&&NewType>. It still reads more cleanly though I think.